code/items/ammo/TTTAmmo.cs

```diff
@@ -50,58 +52,83 @@ public override void Spawn()
             Tags.Add(IItem.ITEM_TAG);
         }
 
-        public override void Touch(Entity other)
+        public void SetCurrentAmmo(int ammo)
         {
-            base.Touch(other);
-
-            if (!IsServer || other is not TTTPlayer player)
-            {
-                return;
-            }
+            CurrentAmmo = ammo;
+        }
 
-            string ammoType = AmmoName.ToLower();
-            Inventory inventory = player.Inventory;
+        public override void TakeDamage(DamageInfo info)
+        {
+            PhysicsBody body = info.Body;
 
-            if (!inventory.GetAmmoTypes().Contains(ammoType))
+            if (!body.IsValid())
             {
-                return;
+                body = PhysicsBody;
             }
 
-            int playerAmount = inventory.Ammo.Count(ammoType);
-
-            if (!(Max >= (playerAmount + Math.Ceiling(CurrentAmmo * 0.25))))
+            if (body.IsValid() && !info.Flags.HasFlag(DamageFlags.PhysicsImpact))
             {
-                return;
+                body.ApplyImpulseAt(info.Position, info.Force * 100);
             }
+        }
 
-            int amountGiven = Math.Min(CurrentAmmo, Max - playerAmount);
-            inventory.Ammo.Give(ammoType, amountGiven);
-            CurrentAmmo -= amountGiven;
-            OnPickup.Fire(other);
+        public float HintDistance => 80f;
 
-            if (CurrentAmmo <= 0 || Math.Ceiling(AmmoEntMax * 0.25) > CurrentAmmo)
-            {
-                Delete();
-            }
+        public TranslationData TextOnTick => new("GENERIC_PICKUP", new object[] { Input.GetKeyWithBinding("+iv_use").ToUpper() });
```

Comment:
We need a way to provide the translated string for the item, is there a way to do this? We would need to use `GetTranslation` @Alf21

Reply:
Already possible <https://github.com/TTTReborn/tttreborn/blob/086d86f86f41b95cc5626a2ff699e17e018aac39/code/lang/Language.cs#L98-L101, e.g. used here <https://github.com/TTTReborn/tttreborn/blob/40bfd72fe15b23a90d1cfe91f5f9b60bf3d54e82/code/ui/generalhud/menu/Menu.ShopEditor.ItemEdit.cs#L37>